lib/ymlr/encoder.ex

```diff
@@ -105,7 +105,18 @@ defmodule Ymlr.Encoder do
   end
 
   defp encode_as_io_list(data, _) when is_atom(data), do: Atom.to_string(data)
-  defp encode_as_io_list(data, level) when is_binary(data) do
+  defp encode_as_io_list(data, level) when is_binary(data), do: encode_binary(data, level)
+
+  defp encode_as_io_list(data, _) when is_number(data), do: "#{data}"
```

Comment:
Is it necessary to change it to binary? Wouldn't io to string take care of that latet if you just returned the number?

Reply:
encode_binary/2 decides when to wrap something in quotes and deals with multi-line strings (indentation)